components/layout_2020/formatting_contexts.rs

```diff
@@ -20,44 +20,43 @@ use style::values::specified::text::TextDecorationLine;
 
 /// https://drafts.csswg.org/css-display/#independent-formatting-context
 #[derive(Debug, Serialize)]
-pub(crate) struct IndependentFormattingContext {
+pub(crate) enum IndependentFormattingContext {
```

Comment:
What do we gain with this change? In particular pushing common fields like `style` further "down" into enum variants.

By the way, in a sizing model as described in https://github.com/w3c/csswg-drafts/issues/5190#issuecomment-643329880 where the parent provides an available size on both axes and the child return its used size of both axis, I think we may be able to unify replace content with other kinds of independent formatting contexts and make it so that callers don’t need to handle them separately.

Reply:
It lets me write a method on `NonReplacedFormattingContext` that will use `StyleExt::outer_inline_and_percentages` and compute the content size on demand. I can't do that without splitting the two kinds of contents given the computation isn't done at all for replaced content.

Reply:
If I didn't do that, I would need to change `NonReplacedIFC` to also have a reference to the style field of `IndependentFormattingContext`, or have a method on `NonReplacedIFC` that takes a style argument, which seems way more confusing to me.